### PR TITLE
fix: escape `$` for fish shells

### LINKED ISSUE #53: Running a single test breaks on fish
This is similar to #50 and also might have a similar fix to https://github.com/nvim-neotest/neotest-go/pull/9.

I am having this error while using fish shell and running `:lua require("neotest").run.run()`:

```bash
fish: $/ is not a valid variable in fish.
cargo nextest run --workspace --no-fail-fast --config-file /var/folders/j5/fyvk84zx4r7fpgfpzs7my4m40000gn/T/nvim.storopoli/B0RHTL/9.nextest.toml --profile neotest --test wallet -E "package(bdk) & test(/^test_bump_fee_reduce_change$/)"
```

We need to scape the `$` sign with `\$`:

```bash
cargo nextest run --workspace --no-fail-fast --config-file /var/folders/j5/fyvk84zx4r7fpgfpzs7my4m40000gn/T/nvim.storopoli/B0RHTL/9.nextest.toml --profile neotest --test wallet -E "package(bdk) & test(/^test_bump_fee_reduce_change\$/)"
warning: some crates are on edition 2021 which defaults to `resolver = "2"`, but virtual workspaces default to `resolver = "1"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
warning: some crates are on edition 2021 which defaults to `resolver = "2"`, but virtual workspaces default to `resolver = "1"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
    Finished test [unoptimized + debuginfo] target(s) in 0.15s
    Starting 1 test across 1 binary (121 skipped)
        FAIL [   0.022s] bdk::wallet test_bump_fee_reduce_change

--- STDOUT:              bdk::wallet test_bump_fee_reduce_change ---

running 1 test
test test_bump_fee_reduce_change ... FAILED

failures:

failures:
    test_bump_fee_reduce_change

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 121 filtered out; finished in 0.02s


--- STDERR:              bdk::wallet test_bump_fee_reduce_change ---
thread 'test_bump_fee_reduce_change' panicked at 'called `Result::unwrap()` on an `Err` value: FeeRateTooLow { required: FeeRate(2.9230769) }', crates/bdk/tests/wallet.rs:1445:33
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

------------
     Summary [   0.023s] 1 test run: 0 passed, 1 failed, 121 skipped
        FAIL [   0.022s] bdk::wallet test_bump_fee_reduce_change
error: test run failed
```

### PR DESCRIPTION
**Pull request by realeinherjar**
This closes #53.

### CODE REVIEW

File: lua/neotest-rust/init.lua [hunk 1]
```diff
@@ -263,7 +263,7 @@ function adapter.build_spec(args)
     if position.type == "test" then
         position_id = position.id
         -- TODO: Support rstest parametrized tests
-        test_filter = '-E "' .. package_filter .. "test(/^" .. position_id .. '\$/)"'
+        test_filter = '-E "' .. package_filter .. "test(/^" .. position_id .. '$/)"'
```
Comment (realeinherjar) [inline lua/neotest-rust/init.lua:266]: What evil sorcery is this?

**Review by rouge8 — CHANGES_REQUESTED** [review]
Hmm I wondered about this when I merged #52, but I tested and it seemed to work... Can you fix the tests?

**Comment by rouge8** [comment]
Actually @realeinherjar can you test #55 and see if that resolves your issue?

**Comment by realeinherjar** [comment]
Closing this in favor of #55

**Pull request closed without merging**